src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,233 @@
+  int overlapEnd ;
+
+  /** Reference sequence in feature (this should be the codon at CDS level) */
+  union { null, string } referenceSequence = null;
+
+  /** Alternate sequence in feature (this should be the codon at CDS level) */
+  union { null, string } alternateSequence = null;
+
+}
+
+/**
+Impact is a simple prioritization for the effect of an allele which is used
+in the annotation record.
+IMPORTANT:
+ Prioritization methods are a crude estimates and are not assumed to be
+ reliable: a 'HIGH' Impact may actually not cause any disruption
+ in protein function or expression.
+*/
+enum Impact {
+ HIGH,      // Variant highly disrupts protein function
+ MODERATE,  // Moderately disrupts protein function
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+An annotation set record groups VariantAnnotation and/or AlleleAnnotation records. It holds
+information describing the software and reference data used in the annotation.
+*/
+record AnnotationSet {
+
+  /** The ID of the annotation set record */
+  string id;
+
+  /** The ID of the dataset to which this annotation set belongs */
+  string datasetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference datasets used.
+  */
+  Analysis analysis;
+
+}
+
+
+/**
+A transcript effect record is a set of information describing the
+effect of an allele on a transcript
+*/
+record TranscriptEffect {
+
+
+  /** The ID of the transcript effect record
+  */
+  string id;
+
+  //TODO: derive unique id from digest of data [location, allele, transcript?]
+
+
+  /** The id of the transcript feature the annotation is relative to
+  */
+  string featureId;
+
+
+  /** Alternate allele - a variant may have more than one alternate allele, each of which
+  will have distinct annotation. This is not be needed for the graph Allele model
+  */
+  union { null, string} alternateBases = null;
+
+  /** Effect of variant on this feature */
+  array<OntologyTerm> effects;
+
+
+  /** Highest Impact from the predicted effects */
+  Impact impact;
+
+
+  /** HGVS formatted annotation at genomic level  */
+  union { null, string } HGVSg = null;
+
+  /** HGVS formatted annotation at transcript level */
+  union { null, string } HGVSc = null;
+
+  /** HGVS formatted annotation at protein level */
+  union { null, string } HGVSp = null;
+
+
+  /** Change relative to cDNA  */
+  union {null, AlleleLocation} cDNALocation =null;
+
+  /* Change relative to coding sequence  */
+  union { null, AlleleLocation} CDSLocation = null;
+
+  /** Change relative to protein  */
+  union { null, AlleleLocation} proteinLocation = null;
+
+
+  /** Output from prediction packages such as Sift */
+  array<AnalysisResult> analysisResults;
+
+
+}
+
+
+
+
+/**
+A `VariantAnnotation` record represents the result of comparing a variant
+to a set of reference data.
+*/
+record VariantAnnotation {
```

Comment:
Yes that's right. There's a new comment at the top of the file to this effect.

I'm aware of the disussion re splitting graph and linear modes into different branches and it won't be hard to split this PR as neccessary.